### DISCUSSION TITLE
Which target should be chosen for ESP32-S2-DevKitC-1?

### DISCUSSION THREAD

**Discussion by jeffer80** [nanoframework/Home#1285] [Q&A] [answered]
I flash the FEATHER_S2 into it, but it can't be found by device explorer.
If I connect ESP32-DevKitC with ESP32_BLE_REV3 to USB, it will be detected immediately. 

details of the devkit:
ESP32-S2 (ESP32-S2R2)
Features WiFi, No Embedded Flash, Embedded PSRAM 2MB, ADC and temperature sensor calibration in BLK2 of efuse V2
Flash size 8MB unknown from ST (manufacturer 0x32 device 0x16407)
PSRAM: available
Crystal 40MHz
MAC 70:04:1D:FA:54:7C

**Comment by alberk8** [comment]
If you use putty with baud rate 115200 , do you see the boot screen and if it goes into a loop.  if so then there is something wrong with the firmware.

**Reply by jeffer80** [reply]
ESP-ROM:esp32s2-rc4-20191025
Build:Oct 25 2019
rst:0x1 (POWERON),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:2
load:0x3ffe6100,len:0x1380
load:0x4004c000,len:0xa58
load:0x40050000,len:0x2ce8
SHA-256 comparison failed:
Calculated: fb5c1027a80b55f8b769710277c85c28c68d0ad3f481d30b54596526b8a84715
Expected: 5ebc2a0d9f72eda82512ece69751d66c0622570153d7d10df14d390dc0ff748a
Attempting to boot anyway...
entry 0x4004c200
I (42) boot: .NET nanoFramework 2nd stage bootloader ESP-IDF v4.4.4
I (42) boot: build Apr 25 2023 22:14:33
I (43) boot: chip revision: v1.0
I (47) boot_comm: chip revision: 1, min. bootloader chip revision: 0
I (54) boot.esp32s2: SPI Speed      : 40MHz
I (59) boot.esp32s2: SPI Mode       : DIO
I (63) boot.esp32s2: SPI Flash Size : 8MB
I (68) boot: Enabling RNG early entropy source...
I (73) boot: Partition Table:
I (77) boot: ## Label            Usage          Type ST Offset   Length
I (84) boot:  0 nvs              WiFi data        01 02 00009000 00006000
I (92) boot:  1 phy_init         RF data          01 01 0000f000 00001000
I (99) boot:  2 factory          factory app      00 00 00010000 001a0000
I (107) boot:  3 deploy           Unknown data     01 84 001b0000 002e0000
I (114) boot:  4 config           Unknown data     01 82 00490000 00200000
I (122) boot: End of partition table
I (126) boot_comm: chip revision: 1, min. application chip revision: 0
I (133) esp_image: segment 0: paddr=00010020 vaddr=3f000020 size=29d3ch (171324)                                                   map
I (188) esp_image: segment 1: paddr=00039d64 vaddr=3ff9e02c size=00004h (     4)                                                   load
I (188) esp_image: segment 2: paddr=00039d70 vaddr=3ffcaa20 size=02f1ch ( 12060)                                                   load
I (197) esp_image: segment 3: paddr=0003cc94 vaddr=40024000 size=03384h ( 13188)                                                   load
I (206) esp_image: segment 4: paddr=00040020 vaddr=40080020 size=b9110h (758032)                                                   map
I (414) esp_image: segment 5: paddr=000f9138 vaddr=40027384 size=13698h ( 79512)                                                   load
I (439) esp_image: segment 6: paddr=0010c7d8 vaddr=40070000 size=0002ch (    44)                                                   load
I (451) boot: Loaded app from partition at offset 0x10000
I (451) boot: Disabling RNG early entropy source...
I (462) cache: Instruction cache        : size 8KB, 4Ways, cache line size 32Byt                                                  e
I (462) cache: Data cache               : size 8KB, 4Ways, cache line size 32Byt                                                  e
I (468) spiram: Found 16MBit SPI RAM device
I (473) spiram: SPI RAM mode: sram 40m
I (477) spiram: PSRAM initialized, cache is in normal (1-core) mode.
I (484) cpu_start: Pro cpu up.
I (507) cpu_start: Pro cpu start user code
I (507) cpu_start: cpu freq: 240000000
I (507) cpu_start: Application information:
I (510) cpu_start: Project name:     nanoCLR
I (515) cpu_start: App version:      1.8.1.166
I (520) cpu_start: Compile time:     Apr 25 2023 22:13:58
I (526) cpu_start: ELF file SHA256:  8d6a5f99317bcaf4...
I (532) cpu_start: ESP-IDF:          v4.4.4
I (537) heap_init: Initializing. RAM available for dynamic allocation:
I (544) heap_init: At 3FFD7000 len 00025000 (148 KiB): DRAM
I (550) heap_init: At 3FFFC000 len 00003A10 (14 KiB): DRAM
I (556) heap_init: At 3FF9E030 len 00001FD0 (7 KiB): RTCRAM
I (563) spiram: Adding pool of 2048K of external SPI memory to heap allocator
I (571) spi_flash: detected chip: generic
I (575) spi_flash: flash io: dio
I (591) cpu_start: Starting scheduler on PRO CPU.
I (591) spiram: Reserving pool of 32K of internal memory for DMA/internal allocations

**Comment by alberk8** [comment]
Can you plug in the the other USB?

**Reply by jeffer80** [reply]
same in any usb port, even I tryied on other computer, the extension did not work also.

**Comment by jeffer80** [comment]
I can deploy my image by using nanoff, and it runs successfully.
and some warning messages showed when flashing the image.
is the same reason why the extension can not detect the devkit?

`C:\Users\jeffe>nanoff --target FEATHER_S2 --serialport COM9 --deploy --image D:\Development\Jeffer\biaer\iot\esp32-client\bin\Debug\Esp32Client.bin
.NET nanoFramework Firmware Flasher v2.4.34+09d44f5033
Copyright (C) 2019 .NET Foundation and nanoFramework project contributors

** Can't check the version! **
** Continuing anyway. **

Using COM9 @ 1500000 baud to connect to ESP32.
Reading details from chip...
OK`

**Reply by alberk8** [reply]
Your kit has two usb port, which one are you using?

**Reply by jeffer80** [reply]
Thanks, it works!
I used the USB-UART port, and it works on the other one.
I'm a newbie, thanks.

there is another question:
The 'System.Device.UsbStream' is not supported by the FEATHER_S2  target, 
so can I use Esp3-S2 as a usb client? 
if yes, how can I do?

**Reply by alberk8** [reply, marked as answer]
I don't think the USBStream is implemented in ESP32 yet.

**Reply by josesimoes** [reply]
That is correct. Currently UsbStream is only supported in Silabs MCUs.
